File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/bookmark/repository/BookmarkRepository.java
package io.dustin.apps.board.domain.bookmark.repository;

import io.dustin.apps.board.domain.bookmark.model.Bookmark;
import io.dustin.apps.common.repository.BaseRepository;

import java.util.Optional;

public interface BookmarkRepository extends BaseRepository<Bookmark, Long> {

    void deleteByUserIdAndBoardId(Long userId, Long boardId);

    Optional<Bookmark> findByUserIdAndBoardId(Long userId, Long boardId);
}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/bookmark/repository/BookmarkRepository.kt
package io.dustin.apps.board.domain.bookmark.repository

import io.dustin.apps.board.domain.bookmark.model.Bookmark
import io.dustin.apps.common.repository.BaseRepository
import java.util.*

interface BookmarkRepository : BaseRepository<Bookmark, Long> {
    fun deleteByUserIdAndBoardId(userId: Long, boardId: Long)
    fun findByUserIdAndBoardId(userId: Long, boardId: Long): Bookmark?
}
